### PR TITLE
Fix: use NODE_BINARY when starting packager

### LINKED ISSUE #22868: packager.sh seems to not find node binary when using NODE_BINARY variable
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment

```
  React Native Environment Info:
    System:
      OS: Linux 4.19 Fedora 29 (Twenty Nine) 29 (Twenty Nine)
      CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
      Memory: 3.14 GB / 15.55 GB
      Shell: 4.4.23 - /bin/bash
    Binaries:
      Node: 11.4.0 - ~/.local/share/virtualenvs/luko-app-react-native-wkPbUArk/bin/node
      npm: 6.5.0 - ~/.local/share/virtualenvs/luko-app-react-native-wkPbUArk/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      Android SDK:
        API Levels: 23, 26, 27, 28
        Build Tools: 27.0.3, 28.0.3
        System Images: android-27 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
    npmPackages:
      react: ^16.6.3 => 16.7.0 
      react-native: ^0.57.8 => 0.57.8 
```

## Description

In `https://github.com/facebook/react-native/blob/master/scripts/packager.sh` node is called directly, without using the one defined by `NODE_BINARY` variable

So, when using other alternatives than nvm or nodenv (I use nodeenv, but in a pipenv python virtualenv, to be fully isolated from the system), XCode fail, telling node is not found

## Reproducible Demo
To setup a virtualenv as I did : 

Install python pipenv tool at user level:
```
pip install -u pipenv
```
Then setup a virtualenv and activate node
```
cd my_src_directory
pipenv --python 3
pipenv shell
pipenv install --dev nodeenv
nodeenv -p
```
Setup the react-native stack
```
npm install
```
Retrieve the virtualenv path
```
pipenv --venv
```
Configure xcode shell script for react native building with NODE_BINARY using the virtualenv path (+ bin/ subdirectory) then try to build => Error

### PR DESCRIPTION
**Pull request by Esemesek**
## Summary

Fix packager script to use `NODE_BINARY` env variable.

Should fix #22868 

## Changelog

[iOS] [Fixed] - Use `NODE_BINARY` env variable in `packager.sh` script

## Test Plan

A launched app in Debug and Release mode and running `packager.sh` directly from `node_modules`.

cc. @Nabellaleen - since I wasn't able to reproduce the crash, can you verify that this does work in your project?

### CODE REVIEW

File: scripts/node-binary.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+[ -z "$NODE_BINARY" ] && export NODE_BINARY="node"
+
+nodejs_not_found()
+{
+  echo "error: Can't find '$NODE_BINARY' binary to build React Native bundle" >&2
```
Comment (cpojer) [inline scripts/node-binary.sh]: This should end in a "."

File: scripts/node-binary.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  echo "If you have non-standard nodejs installation, select your project in Xcode," >&2
```
Comment (cpojer) [inline scripts/node-binary.sh]: "If you have a non-standard node.js […]"

File: scripts/node-binary.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  echo "find 'Build Phases' - 'Bundle React Native code and images'" >&2
+  echo "and change NODE_BINARY to absolute path to your node executable" >&2
```
Comment (cpojer) [inline scripts/node-binary.sh]: "to an absolute path to your […]"

Comment (cpojer) [reply scripts/node-binary.sh]: End with a "."

File: scripts/node-binary.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  echo "(you can find it by invoking 'which node' in the terminal)" >&2
```
Comment (cpojer) [inline scripts/node-binary.sh]: This should be a separate sentence without parenthesis.

**Comment by Nabellaleen** [comment]
@Esemesek : I take time to check this tomorrow !

**Pull request closed without merging**